packages/flutter/lib/src/widgets/scrollbar.dart

```diff
@@ -386,3 +401,536 @@ class ScrollbarPainter extends ChangeNotifier implements CustomPainter {
+    super.initState();
+    _fadeoutAnimationController = AnimationController(
+      vsync: this,
+      duration: widget.fadeDuration,
+    );
+    _fadeoutOpacityAnimation = CurvedAnimation(
+      parent: _fadeoutAnimationController,
+      curve: Curves.fastOutSlowIn,
+    );
+  }
+
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+    // If a painter has not been created by a subclass, create a raw scrollbar
+    // painter.
+    painter ??= _buildRawScrollbarPainter();
+    maybeTriggerScrollbar();
+  }
+
+  /// Waits one frame and cause an empty scroll event.
+  ///
+  /// This allows the thumb to show immediately when isAlwaysShown is true.
+  /// A scroll event is required in order to paint the thumb.
+  @protected
+  void maybeTriggerScrollbar() {
+    WidgetsBinding.instance!.addPostFrameCallback((Duration duration) {
+      if (widget.isAlwaysShown) {
+        _fadeoutTimer?.cancel();
+        // Wait one frame and cause an empty scroll event.  This allows the
+        // thumb to show immediately when isAlwaysShown is true. A scroll
+        // event is required in order to paint the thumb.
+        widget.controller!.position.didUpdateScrollPositionBy(0);
+      }
+    });
+  }
+
+  ScrollbarPainter _buildRawScrollbarPainter() {
+    return ScrollbarPainter(
+      color: widget.thumbColor,
+      textDirection: Directionality.of(context),
+      thickness: widget.thickness,
+      radius: widget.radius,
+      fadeoutOpacityAnimation: fadeoutOpacityAnimation,
+      padding: MediaQuery.of(context).padding,
+    );
+  }
+
+  @override
+  void didUpdateWidget(T oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    if (widget.isAlwaysShown != oldWidget.isAlwaysShown) {
+      if (widget.isAlwaysShown == true) {
+        maybeTriggerScrollbar();
+        _fadeoutAnimationController.animateTo(1.0);
+      } else {
+        _fadeoutAnimationController.reverse();
+      }
+    }
+    painter!
+      ..thickness = widget.thickness
+      ..radius = widget.radius
+      ..color = widget.thumbColor;
+  }
+
+  void _updateScrollPosition(double primaryDelta) {
+    assert(_currentController != null);
+
+    // Convert primaryDelta, the amount that the scrollbar moved since the last
+    // time _dragScrollbar was called, into the coordinate space of the scroll
+    // position, and jump to that position.
+    final double scrollOffsetLocal = painter!.getTrackToScroll(primaryDelta);
+    final double scrollOffsetGlobal = scrollOffsetLocal + _currentController!.position.pixels;
+    _currentController!.position.jumpTo(scrollOffsetGlobal);
+  }
+
+  void _maybeStartFadeoutTimer() {
+    if (!widget.isAlwaysShown) {
+      _fadeoutTimer?.cancel();
+      _fadeoutTimer = Timer(widget.timeToFade, () {
+        _fadeoutAnimationController.reverse();
+        _fadeoutTimer = null;
+      });
+    }
+  }
+
+  /// Returns the [Axis] of the child scroll view, or null if one is not
+  /// determined.
```

Comment:
What situation would lead to a null return value?  In other words do callers have to unconditionally guard against a null return value or are there circumstances where it's safe to assume that the return value is non-null?